core/src/main/java/org/wildfly/channel/ChannelSession.java

```diff
@@ -55,10 +56,16 @@ public ChannelSession(List<Channel> channels, MavenVersionsResolver.Factory fact
      * @param artifactId - required
      * @param extension - can be null
      * @param classifier - can be null
+     * @param baseVersion - required
```

Comment:
We have a case where the version can be null and expected to be resolved by the channel. When a verion is expressed as an expression, for example: https://github.com/jfdenise/eap-datasources-galleon-pack/blob/main/galleon-feature-pack/src/main/resources/resources/wildfly/artifact-versions.properties#L2
If the expression can't be resolved and channels have been configured, then the version is null and is expected to be resolved by the channel library.